src/Core/SonarLintNotifications/INotificationSettingStorage.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarLint.VisualStudio.Core.SonarLintNotifications
+{
+    public enum StorageLocation
+    {
+        User,
+        VSInstance,
+        Solution
+    }
+
+    public interface INotificationSettingStorage
+    {
+        void SaveNotificationSetting(INotificationSetting setting, StorageLocation storageLocation);
+        void GetNotificationSetting(string notificationSettingKey, StorageLocation storageLocation);
```

Comment:
The `get` doesn't return anything.

Reply:
The caller has to know where the data is stored or they won't find anything. I guess it doesn't matter here, since in the expected usage there will only be one caller using each notification key, and they will be calling `get` then `set`.